pkg/ds/tripleset_outer.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ds
+
+// OuterTripleSet is an outer-associative 3-product of sets (S1 x S3) x S2
+type OuterTripleSet[S1 Set[S1], S2 Set[S2], S3 Set[S3]] struct {
```

Comment:
why do we need `OuterTripleSet` and `RightTripleSet` ?

Reply:
It implies different partitions.